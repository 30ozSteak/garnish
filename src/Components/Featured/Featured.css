.featured-container .scroll-text {
  text-align: center;
  font-weight: 300;
  color: gray;
  animation: fadeIn 2s linear infinite;
  position: relative;
}

.featured-box {
  display: flex;
  border: none;
  height: 13rem;
  margin: 0 auto;
  overflow: scroll;
}

.channel-box {
  padding: 1rem;
  height: 12.5rem;
}

.channel-button {
  width: 18rem;
  height: 10rem;
  background-color: white;
  border-radius: 2px;
  box-shadow: 5px 5px 20px lightgray;
  transition: 0.2s;
  overflow: hidden;
}

.channel-button:hover {
  box-shadow: 5px 5px 20px gray;
  cursor: pointer;
}

.channel-button p {
  font-size: 4.3rem;
  font-weight: 600;
  padding: 1.5rem 0.5rem 0 1rem;
}

.channel-button .channel-desc {
  font-size: 0.8rem;
  font-weight: 300;
}
