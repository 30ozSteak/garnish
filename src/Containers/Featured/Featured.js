import React from "react";
import { connect } from "react-redux";
import { getChannel, activeChannel } from "../actions";

const Channel = ({ channelName, channelString, onClick, active }) => (
  <div onClick={onClick} className="featured -box">
    <div className="featured-button">
      <p>{channelName}</p>
    </div>
  </div>
);

export const mapStateToProps = state => ({
  active: state.channel
});

export const mapDispatchToProps = (dispatch, ownProps) => ({
  onClick: () => {
    dispatch(getChannel(ownProps.channelString));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Channel);
