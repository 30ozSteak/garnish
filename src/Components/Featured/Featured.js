import React from "react";
import "./Featured.css";
import Channels from "../../Containers/Channels/Channels";

const Featured = () => (
  <div className="featured-container">
    <div className="featured-box">
      <Channels channelName="Tech" channelString="Tech" />
      <Channels channelName="Memes" channelString="Tech" />
      <Channels channelName="Web Dev" channelString="Tech" />
      <Channels channelName="Tech" channelString="Tech" />
      <Channels channelName="Tech" channelString="Tech" />
    </div>
  </div>
);

export default Featured;
