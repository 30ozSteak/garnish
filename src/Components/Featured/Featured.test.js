import React from "react";
import Featured from "./Featured";
import { shallow } from "enzyme";

describe("Featured", () => {
  it("should resemble the snapshot", () => {
    const wrapper = shallow(<Featured />);
    expect(wrapper).toMatchSnapshot();
  });
});
