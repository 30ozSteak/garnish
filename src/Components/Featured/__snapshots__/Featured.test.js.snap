// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured should resemble the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Featured />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="featured-box"
        >
          <Channels
            channelName="tech"
            channelString="new in tech, today"
            type="tech"
          />
          <Channels
            channelName="memes"
            channelString="fortnite floss, kids falling over, the works"
            type="memes"
          />
          <Channels
            channelName="abramov"
            channelString="whatever dan says"
            type="abramov"
          />
          <Channels
            channelName="culture"
            channelString="learn some things"
            type="culture"
          />
          <Channels
            channelName="top news"
            channelString="for when you want to be sad"
            type="top"
          />
        </div>,
        <p
          className="scroll-text"
        >
          scroll
        </p>,
      ],
      "className": "featured-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Channels
              channelName="tech"
              channelString="new in tech, today"
              type="tech"
            />,
            <Channels
              channelName="memes"
              channelString="fortnite floss, kids falling over, the works"
              type="memes"
            />,
            <Channels
              channelName="abramov"
              channelString="whatever dan says"
              type="abramov"
            />,
            <Channels
              channelName="culture"
              channelString="learn some things"
              type="culture"
            />,
            <Channels
              channelName="top news"
              channelString="for when you want to be sad"
              type="top"
            />,
          ],
          "className": "featured-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "channelName": "tech",
              "channelString": "new in tech, today",
              "handleNews": undefined,
              "type": "tech",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "channelName": "memes",
              "channelString": "fortnite floss, kids falling over, the works",
              "handleNews": undefined,
              "type": "memes",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "channelName": "abramov",
              "channelString": "whatever dan says",
              "handleNews": undefined,
              "type": "abramov",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "channelName": "culture",
              "channelString": "learn some things",
              "handleNews": undefined,
              "type": "culture",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "channelName": "top news",
              "channelString": "for when you want to be sad",
              "handleNews": undefined,
              "type": "top",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "scroll",
          "className": "scroll-text",
        },
        "ref": null,
        "rendered": "scroll",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="featured-box"
          >
            <Channels
              channelName="tech"
              channelString="new in tech, today"
              type="tech"
            />
            <Channels
              channelName="memes"
              channelString="fortnite floss, kids falling over, the works"
              type="memes"
            />
            <Channels
              channelName="abramov"
              channelString="whatever dan says"
              type="abramov"
            />
            <Channels
              channelName="culture"
              channelString="learn some things"
              type="culture"
            />
            <Channels
              channelName="top news"
              channelString="for when you want to be sad"
              type="top"
            />
          </div>,
          <p
            className="scroll-text"
          >
            scroll
          </p>,
        ],
        "className": "featured-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Channels
                channelName="tech"
                channelString="new in tech, today"
                type="tech"
              />,
              <Channels
                channelName="memes"
                channelString="fortnite floss, kids falling over, the works"
                type="memes"
              />,
              <Channels
                channelName="abramov"
                channelString="whatever dan says"
                type="abramov"
              />,
              <Channels
                channelName="culture"
                channelString="learn some things"
                type="culture"
              />,
              <Channels
                channelName="top news"
                channelString="for when you want to be sad"
                type="top"
              />,
            ],
            "className": "featured-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "channelName": "tech",
                "channelString": "new in tech, today",
                "handleNews": undefined,
                "type": "tech",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "channelName": "memes",
                "channelString": "fortnite floss, kids falling over, the works",
                "handleNews": undefined,
                "type": "memes",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "channelName": "abramov",
                "channelString": "whatever dan says",
                "handleNews": undefined,
                "type": "abramov",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "channelName": "culture",
                "channelString": "learn some things",
                "handleNews": undefined,
                "type": "culture",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "channelName": "top news",
                "channelString": "for when you want to be sad",
                "handleNews": undefined,
                "type": "top",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "scroll",
            "className": "scroll-text",
          },
          "ref": null,
          "rendered": "scroll",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
